packages/f/freetype/xmake.lua

```diff
@@ -54,27 +54,26 @@ package("freetype")
         local configs = {}
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
         table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
-        local function add_dep(dep, cmakeConf, cmakeDisableConf)
-            if package:config(dep) then
-                table.insert(configs, "-DFT_WITH_" .. cmakeConf .. "=ON")
-            else
-                table.insert(configs, "-DCMAKE_DISABLE_FIND_PACKAGE_" .. (cmakeDisableConf or cmakeConf) .. "=ON")
-            end
-        end
-        add_dep("bzip2", "BZIP2", "BZip2")
-        add_dep("png", "PNG")
-        add_dep("woff2", "BROTLI", "BrotliDec")
-        add_dep("zlib", "ZLIB")
+        local function add_dep(opt)
+            if package:config(opt.conf) then
+                table.insert(configs, "-DFT_WITH_" .. opt.cmakewith .. "=ON")
 
-        -- brotli isn't found automatically
-        if package:config("woff2") then
-            local brotli = package:dep("brotli")
-            if brotli and not brotli:is_system() then
-                local fetchinfo = brotli:fetch()
-                table.insert(configs, "-DBROTLIDEC_INCLUDE_DIRS=" .. table.concat(fetchinfo.includedirs or fetchinfo.sysincludedirs, ";"))
-                table.insert(configs, "-DBROTLIDEC_LIBRARIES=" .. table.concat(fetchinfo.libfiles, ";"))
+                local lib = package:dep(opt.pkg or opt.conf)
+                if lib and not lib:is_system() then
+                    local includeconf = opt.cmakeinclude or (opt.cmakewith .. "_INCLUDE_DIRS")
+                    local libconf = opt.cmakelib or (opt.cmakewith .. "_LIBRARIES")
+                    local fetchinfo = lib:fetch()
```

Comment:
I think you should add `if fetchinfo then` or assert

Reply:
In what cases can :fetch() return nil in this case?

Reply:
> In what cases can :fetch() return nil in this case?

Normally, it will not return nil, but it will always be fine after checking.